src/Storages/HDFS/StorageHDFS.cpp

```diff
@@ -72,8 +72,10 @@ namespace
 
         HDFSFileInfo ls;
         ls.file_info = hdfsListDirectory(fs.get(), prefix_without_globs.data(), &ls.length);
-        if (ls.file_info == nullptr) {
-            throw Exception(ErrorCodes::ACCESS_DENIED, "Cannot list directory {}: {}", prefix_without_globs, String(hdfsGetLastError()));
+        if (ls.file_info == nullptr && errno != ENOENT) {
```

Comment:
May be it is better to assume that ls.length is 0 if ls.file_info is nullptr to avoid this errno checks?

Reply:
failed on check style, I have no ideas.
I spend much time finding the bug(https://github.com/ClickHouse-Extras/libhdfs3/pull/18) last week because all type of exception has been ignored

Reply:
libhdfs doesn't have a good way to get the type of exception that occurred. check style doesn't seem to allow direct use of errno.Is there any other way to solve this problem.

Reply:
// NO_LINT can help.

Reply:
thanks, I fix it